meteoblue_dataset_sdk/caching/filecache.py

```diff
@@ -0,0 +1,105 @@
+import datetime
+import hashlib
+import json
+import logging
+import os
+import tempfile
+import zlib
+from typing import Union, Optional
+
+import aiofiles
+from aiofiles import os as aios
+
+from .cache import Cache
+
+CACHE_DIR = "mb_cache"
+# 7200s == 2h
+DEFAULT_CACHE_DURATION = 7200
+
+
+class FileCache(Cache):
+    def __init__(
+        self,
+        cache_path: Optional[str] = None,
+        cache_ttl: int = DEFAULT_CACHE_DURATION,
+        compression_level: int = 6,
+    ):
+        if cache_path is None:
+            cache_path = tempfile.gettempdir()
+        cache_path = os.path.join(cache_path, CACHE_DIR)
+        if not os.path.exists(cache_path):
+            os.makedirs(cache_path)
+
+        self.cache_path = cache_path
+        self.cache_ttl = cache_ttl
+        self.compression_level = compression_level
+
+    async def set(self, query_params: dict, value: bytes) -> None:
```

Comment:
`query_params ` should be renamed to `key`. The terms key value are well established for caches

Reply:
a dict as a key is not very explicit, therefore i preferred using query_param as this, over all implementations must be used

Reply:
The cache interface should not receive the query_params as a dict, but a cache key as a string. In a earlier version this was alreadyy the case.

Please move the query_params to hash serialisation outside of the cache